databricks/koalas/indexes.py

```diff
@@ -2220,20 +2220,20 @@ def intersection(self, other) -> "Index":
         if isinstance(other, DataFrame):
             raise ValueError("Index data must be 1-dimensional")
         elif isinstance(other, MultiIndex):
-            # Always returns an empty MultiIndex if `other` is MultiIndex.
-            return other.to_frame().head(0).index
+            # Always returns a no-named empty Index if `other` is MultiIndex.
+            return self._kdf.head(0).index.rename(None)
```

Comment:
The behavior of `Index.intersection` was changed in pandas 1.2 when the `other` is `MultiIndex`.

It returned empty `MultiIndex` before, but it returns empty `Index` now.